utils/rpms/daoscov.spec

```diff
@@ -0,0 +1,705 @@
+NVM express (NVMe) while presenting a key-value storage interface and
+providing features such as transactional non-blocking I/O, advanced
+data protection with self healing on top of commodity hardware, end-
+to-end data integrity, fine grained data control and elastic storage
+to optimize performance and cost.
+
+%package server
+Summary: The DAOS server
+Requires: %{name}%{?_isa} = %{version}-%{release}
+Requires: %{name}-client%{?_isa} = %{version}-%{release}
+Requires: spdk-tools
+Requires: ndctl
+# needed to set PMem configuration goals in BIOS through control-plane
+%if (0%{?suse_version} >= 1500)
+Requires: ipmctl < 02.00.00.3809
+%else
+Requires: ipmctl > 02.00.00.3816
+%endif
+Requires: hwloc
+Requires: mercury = %{mercury_version}
+Requires(post): /sbin/ldconfig
+Requires(postun): /sbin/ldconfig
+Requires: libfabric >= %{libfabric_version}
+%{?systemd_requires}
+Obsoletes: cart < 1000
+
+%description server
+This is the package needed to run a DAOS server
+
+%package client
+Summary: The DAOS client
+Requires: %{name}%{?_isa} = %{version}-%{release}
+Requires: mercury = %{mercury_version}
+Requires: libfabric >= %{libfabric_version}
+Requires: fuse3 >= 3.4.2
+Obsoletes: cart < 1000
+%if (0%{?suse_version} >= 1500)
+Requires: libfuse3-3 >= 3.4.2
+%else
+# because our repo has a deprecated fuse-3.x RPM, make sure we don't
+# get it when fuse3 Requires: /etc/fuse.conf
+Requires: fuse < 3, fuse3-libs >= 3.4.2
+%endif
+%{?systemd_requires}
+
+%description client
+This is the package needed to run a DAOS client
+
+%package tests
+Summary: The DAOS test suite
+Requires: %{name}-client%{?_isa} = %{version}-%{release}
+Requires: python-pathlib
+Requires: python-distro
+Requires: python2-tabulate
+Requires: fio
+Requires: dbench
+Requires: lbzip2
+%if (0%{?suse_version} >= 1315)
+Requires: libpsm_infinipath1
+%endif
+
+%description tests
+This is the package needed to run the DAOS test suite
+
+%package devel
+# Leap 15 doesn't seem to be creating dependencies as richly as EL7
+# for example, EL7 automatically adds:
+# Requires: libdaos.so.0()(64bit)
+%if (0%{?suse_version} >= 1500)
+Requires: %{name}-client%{?_isa} = %{version}-%{release}
+Requires: %{name}%{?_isa} = %{version}-%{release}
+%endif
+Summary: The DAOS development libraries and headers
+
+%description devel
+This is the package needed to build software with the DAOS library.
+
+%prep
+%autosetup
+
+%build
+
+%define conf_dir %{_sysconfdir}/daos
+
+scons %{?_smp_mflags}      \
+      --config=force       \
+      --no-rpath           \
+      USE_INSTALLED=all    \
+      CONF_DIR=%{conf_dir} \
+      PREFIX=%{?buildroot} \
+     %{?scons_args}
+
+%install
+scons %{?_smp_mflags}                 \
+      --config=force                  \
+      --no-rpath                      \
+      --install-sandbox=%{?buildroot} \
+      %{?buildroot}%{_prefix}         \
+      %{?buildroot}%{conf_dir}        \
+      USE_INSTALLED=all               \
+      COMPILER=covc                   \
```

Comment:
This should be in the `%build` `scons` args, not the `%install` ones.  But also not in this forked spec file, in the `daos.spec` one.

Reply:
New commit uses conditional macro in daos.spec. 
daoscov.spec removed.
I did an experimental build where covc is specifed in %build and the result is not as expected.
Looking at the log, %install runs scons again and builds the tests. The coverage build with this approach looks correct where the tests is looking for the correct coverage file and the code base seems to be instrumented correctly.

Reply:
> Looking at the log, %install runs scons again and builds the tests.

This is a bug then.  Once `scons` is run once to build everything, `scons install` should not rebuild anything.  This doesn't happen in the [latest master](https://build.hpdd.intel.com/blue/rest/organizations/jenkins/pipelines/daos-stack/pipelines/daos/branches/master/runs/4126/nodes/111/log/?start=0) so must be something that is introduced with this PR.